tests/test_string_accessors.py

```diff
@@ -204,14 +228,30 @@ def test_string_accessors_string_index():
     _check(
         assert_type(idx.str.translate({241: "n"}), "pd.Index[str]"),
     )
+    _check(
+        assert_type(idx.str.translate({241: 240}), "pd.Index[str]"),
+    )
+    trans_table: dict[int, int] = {ord("a"): ord("b")}
+    _check(  # tests covariance of table values (table is read-only)
+        assert_type(idx.str.translate(trans_table), "pd.Index[str]"),
+    )
     _check(assert_type(idx.str.upper(), "pd.Index[str]"))
     _check(assert_type(idx.str.wrap(80), "pd.Index[str]"))
     _check(assert_type(idx.str.zfill(10), "pd.Index[str]"))
     idx_bytes = pd.Index([b"a1", b"b2", b"c3"])
     _check(assert_type(idx_bytes.str.decode("utf-8"), "pd.Index[str]"))
+    _check(
+        assert_type(
+            idx_bytes.str.decode("utf-8", dtype=pd.StringDtype()), "pd.Index[str]"
+        )
+    )
     idx_list = pd.Index([["apple", "banana"], ["cherry", "date"], ["one", "eggplant"]])
     _check(assert_type(idx_list.str.join("-"), "pd.Index[str]"))
 
+    # wrap doesn't accept positional arguments other than width
+    with pytest.raises(TypeError):
+        idx.str.wrap(80, False)  # type: ignore[misc] # pyright: ignore[reportCallIssue]
```

Comment:
I would usually recommend you use `TYPE_CHECKING_INVALID_USAGE` instead of wrapping it in a pytest context manager: the goal is to test the type checking, not the error raised (same for the other places).

Reply:
Sure happy to change it but isn't `pytest.raises` better because it tests the thing we are claiming to raise actually raises?

Reply:
It depends what you want to test, the type checker has nothing to do with pytest, what we want is to make sure that pyright and mypy complain thus you put `# type: ignore[misc] # pyright: ignore[reportCallIssue]`.
The fact that it raises a TypeError is something that pandas controls, not the stubs. That is why we wrap it in `TYPE_CHECKING_INVALID_USAGE` so it is not run when testing but we only check how mypy and pyright behave.

Reply:
One other note is that if one day pandas decides to change the TypeError into for example a RuntimeError, you would have to change the test but fundamentally the behavior of the types has not changed. That is why it is often easier for future maintenance to wrap it with `TYPE_CHECKING_INVALID_USAGE` instead of a `pytest.raises`, maybe @Dr-Irv can add some details if I missed something.

Reply:
> One other note is that if one day pandas decides to change the TypeError into for example a RuntimeError, you would have to change the test but fundamentally the behavior of the types has not changed. That is why it is often easier for future maintenance to wrap it with `TYPE_CHECKING_INVALID_USAGE` instead of a `pytest.raises`, maybe @Dr-Irv can add some details if I missed something.

@loicdiridollou is correct.  We are just testing the type checking so that's why we do it this way